### DISCUSSION TITLE
Hard to get a stack trace

### DISCUSSION THREAD

**Discussion by SimonsUnikaNamn** [pmndrs/jotai#2749] [Q&A]
Hi, 

I am having a bit of a nasty bug where the inputted parameter does not adhere to the value in the state.

In recoil I could just put a `debugger` line inside a `if` statement in the `atomFamily` that is called with the inconsistent values and look at the stack trace in my dev tool to find the correct path.
In Jotai this is not possible

<img width="220" alt="Skärmavbild 2024-09-25 kl  14 07 04" src="https://github.com/user-attachments/assets/0df4e018-c464-4f5f-b8f0-6de646151f08">

Apologies for the swedish text, but I guess you can see the issue. The stack trace should be several layers, but it is only 2 functions deep, which is strange to me. In fact, the `App.tsx` is not even calling the `atomFamily` in `features.tsx`

Is it possible to fix this?

**Comment by dai-shi** [comment]
Thanks for opening a discussion.
I'm not sure if it's fixable. tbh, I'm not super familiar with stacktrace, so if anyone can give some hints, I would appreciate it.

Maybe not related, but we have an old issue in the past: #931

**Comment by SimonsUnikaNamn** [comment]
Not sure, in recoil I would get something looking like this when stopping with a `debugger`

`DeeplyNestedAtom.tsx:123`
`MoreShallowAtomCallingDeeplyNestedAtom.tsx:64`
`EvenMoreShallowAtomCallingMoreShallowAtomCallingDeeplyNestedAtom.tsx:64`
`...`

So I could trace it all the way up the tree. I don't get that with jotai for some reason or other

I don't think that one is related, I saw it before, but maybe should look again

**Comment by himself65** [comment]
I think there're lots of anonymous function in jotai source code, wonder if we repalce with the normal function?

**Reply by dai-shi** [reply]
I see. I prefer allow functions to make the bundle size smaller, but if that's causing this debugging issue, we could use named function expressions. Would anyone like to try that and see if it improves the stack trace?

> In fact, the `App.tsx` is not even calling the `atomFamily` in `features.tsx`

I wonder if it's related with anonymous functions in the Jotai core, though.

**Reply by himself65** [reply]
from bundler perspective, I don't think there will have very significant size difference if you enable minify?

<img width="161" alt="image" src="https://github.com/user-attachments/assets/fa0633ad-965f-4226-a04a-528f0f9d4f15">

**Reply by dai-shi** [reply]
true for function declaration vs. const arrow function. another reason is arrow function discourage the use of `this` explicitly.
anyway, my question is if it actually fixes OP's concern.
